p2p/transport/tcp/tcp.go

```diff
@@ -186,15 +181,17 @@ func (mt *MultiplexTransport) Dial(
 		return nil, err
 	}
 
-	secretConn, err := mt.upgrade(c, &addr)
+	mconn, _, err := mt.upgrade(c, &addr)
 	if err != nil {
 		return nil, err
 	}
+	mconn.SetLogger(mt.logger.With("remote", addr))
+
+	go mt.cleanupConn(c.RemoteAddr(), mconn.Quit())
 
-	return secretConn, nil
+	return mconn, mconn.Start()
```

Comment:
Maybe we should not start the `mconn` here, but when the peer starts.

In this way we can call `OpenStream` before the multiplex connection is started, avoiding locks around the streams table.